linera-storage/src/migration.rs

```diff
@@ -0,0 +1,572 @@
+    BlobState(BlobId),
+    Event(EventId),
+    BlockExporterState(u32),
+    NetworkDescription,
+}
+
+const UNUSED_EMPTY_KEY: &[u8] = &[];
+// We choose the ordering of the variants in `BaseKey` and `RootKey` so that
+// the root keys corresponding to `ChainState` and `BlockExporter` remain the
+// same in their serialization.
+// This implies that data on those root keys do not need to be moved.
+// For other tag variants, that are on the shared partition, we need to move
+// the data into their own partitions.
+const MOVABLE_KEYS_0_1: &[u8] = &[1, 2, 3, 4, 5, 7];
+
+/// The total number of keys being migrated in a block.
+/// we use chunks to avoid OOM
+const BLOCK_KEY_SIZE: usize = 90;
+
+// We map a `BaseKey` in the shared partition to a `(RootKey,key)` in the
+// new schema. For `ChainState` and `BlockExporterState`, there is no need
+// to move so we use `UNUSED_EMPTY_KEY`.
+//
+fn map_base_key(base_key: &[u8]) -> Result<(Vec<u8>, Vec<u8>), ViewError> {
+    let base_key = bcs::from_bytes::<BaseKey>(base_key)?;
+    match base_key {
+        BaseKey::ChainState(chain_id) => {
+            let root_key = RootKey::ChainState(chain_id).bytes();
+            Ok((root_key, UNUSED_EMPTY_KEY.to_vec()))
+        }
+        BaseKey::Certificate(hash) => {
+            let root_key = RootKey::CryptoHash(hash).bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+        BaseKey::ConfirmedBlock(hash) => {
+            let root_key = RootKey::CryptoHash(hash).bytes();
+            Ok((root_key, ONE_KEY.to_vec()))
+        }
+        BaseKey::Blob(blob_id) => {
+            let root_key = RootKey::Blob(blob_id).bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+        BaseKey::BlobState(blob_id) => {
+            let root_key = RootKey::Blob(blob_id).bytes();
+            Ok((root_key, ONE_KEY.to_vec()))
+        }
+        BaseKey::Event(event_id) => {
+            let root_key = RootKey::Event(event_id.chain_id).bytes();
+            let key = to_event_key(&event_id);
+            Ok((root_key, key))
+        }
+        BaseKey::BlockExporterState(index) => {
+            let root_key = RootKey::BlockExporterState(index).bytes();
+            Ok((root_key, UNUSED_EMPTY_KEY.to_vec()))
+        }
+        BaseKey::NetworkDescription => {
+            let root_key = RootKey::NetworkDescription.bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+    }
+}
+
+impl<Database, C> DbStorage<Database, C>
+where
+    Database: KeyValueDatabase + Clone + Send + Sync + 'static,
+    Database::Store: KeyValueStore + Clone + Send + Sync + 'static,
+    C: Clock + Clone + Send + Sync + 'static,
+    Database::Error: From<bcs::Error> + Send + Sync,
+{
+    async fn migrate_shared_partition(
+        &self,
+        first_byte: &u8,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<(), ViewError> {
+        tracing::info!(
+            "migrate_shared_partition for {} keys starting with with {first_byte}",
```

Comment:
INFO-level logs are meant for the end user, so let's make them a bit easier to understand without knowledge of the code: explain that this is migrating the database format from version 0 to 1, etc. But in this case, maybe this should be DEBUG? The user doesn't care about the first byte.

Or if we keep the INFO log here, maybe we can print the actual data type instead of the first byte? I.e. print that this is now migrating the blobs, blob states, etc.?